### PR TITLE
[3.3.5] Core/Quest: A life without regret

### PR DESCRIPTION
**Pull request by JamminWidYou**
https://github.com/TrinityCore/TrinityCore/issues/5988

todo:
- [X] Spawn creatures next to bank and script short event with Overlord Runthak
- [X] Spawn thrall, sylvanas and script short event with them

https://www.youtube.com/watch?v=DrVajpz3G14
http://www.wowhead.com/quest=13266

### CODE REVIEW

File: sql/updates/world/life_without_regrets.sql [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+(59062, 14, 13257, 13267, 0, 690, 2, 1, 2, 11),
+(59062, 1637, 13257, 13267, 0, 690, 2, 1, 2, 11);
+
+DELETE FROM `spell_script_names` WHERE `spell_id` = @OG_TELEPORTSPELL;
+INSERT INTO `spell_script_names` VALUES
+(@OG_TELEPORTSPELL, 'spell_59064_portal_orgrimmar');
+
+-- Its not sniff verified, if anyone got any info about it, you're more than welcome to correct it
+DELETE FROM `spell_target_position` WHERE `ID` = 58419;
+INSERT INTO `spell_target_position` VALUES
```
Comment (tkrokli) [inline sql/updates/world/life_without_regrets.sql]: The next time you update your source, you need to insert a LineFeed or NewLine at the end of this file. (I put this comment on a line above in case you don't add more lines to this, so this gets overwritten when you update the file.)

**Comment by msoky** [comment]
its quest or what?

**Comment by tkrokli** [comment]
This is about the Horde quest and its events in "A Life Without Regret" which was only available from patch 3.3.0, removed in patch 4.0.3a: http://wow.gamepedia.com/Quest:A_Life_Without_Regret
http://www.wowhead.com/quest=13266/a-life-without-regret
It leads to the next quest http://wow.gamepedia.com/Quest:The_Battle_For_The_Undercity_(Horde) / http://www.wowhead.com/quest=13267/the-battle-for-the-undercity

It is phased, so only those who are in the exact same part of that quest would be visible to each other. I think it is meant to be a single-user quest scenario, although I don't know 100%. I only remember doing it alone on one or two characters back in the winter season of 2009-2010.

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 2]
```diff
@@ -263,8 +264,229 @@ class npc_thrall_warchief : public CreatureScript
 
 };
```
Comment (JamminWidYou) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Can i do this like this?

File: sql/updates/world/life_without_regrets.sql (hunk 1 above)
Comment (JamminWidYou) [inline sql/updates/world/life_without_regrets.sql]: will do next time i push something, didnt notice

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 3]
```diff
@@ -263,8 +264,229 @@ class npc_thrall_warchief : public CreatureScript
...
+    SAY_SANA            = 0, // We demand answers!
+    SAY_RUNTHAK_1       = 0, // SILENCE!
+    SAY_RUNTHAK_2       = 1, // We are on the brink of all out war with the Alliance!
+    SAY_RUNTHAK_3       = 2, // Tragic events have unfolded in Northrend. The Warchief is doing all that he can to keep us safe.
+    SAY_RUNTHAK_4       = 3, // All services and shops are to remain closed until further notice! That is all!
+};
+
+enum misc
+{
+    RUNTHAK_PATH_ID     = 2137980,
```
Comment (Goatform) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Remove comma.

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 4]
```diff
@@ -263,8 +264,229 @@ class npc_thrall_warchief : public CreatureScript
...
+    SAY_OLVIA_2         = 1, // What are all these Forsaken doing here?
+    SAY_FELIKA_1        = 0, // What is going on?
+    SAY_FELIKA_2        = 1, // This is an outrage!
+    SAY_THUNG           = 0, // What is going on?
+    SAY_KAJA            = 0, // Why is Thrall allowing this to happen to our city?
+    SAY_SANA            = 0, // We demand answers!
+    SAY_RUNTHAK_1       = 0, // SILENCE!
+    SAY_RUNTHAK_2       = 1, // We are on the brink of all out war with the Alliance!
+    SAY_RUNTHAK_3       = 2, // Tragic events have unfolded in Northrend. The Warchief is doing all that he can to keep us safe.
+    SAY_RUNTHAK_4       = 3, // All services and shops are to remain closed until further notice! That is all!
```
Comment (Goatform) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Remove comma.

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 5]
```diff
@@ -263,8 +264,229 @@ class npc_thrall_warchief : public CreatureScript
...
+    EVENT_SCENE_5       = 5,
+    EVENT_SCENE_6       = 6,
+    EVENT_SCENE_7       = 7,
+    EVENT_SCENE_8       = 8,
+    EVENT_SCENE_9       = 9,
+    EVENT_SCENE_10      = 10,
+    EVENT_SCENE_11      = 11,
+    EVENT_SCENE_12      = 12,
+    EVENT_SCENE_13      = 13,
+    EVENT_RESET         = 14,
```
Comment (Goatform) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Remove comma.

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 6]
```diff
@@ -263,8 +264,229 @@ class npc_thrall_warchief : public CreatureScript
...
 
+/* --------- Herald of War ------------- */
+enum citizenEntries
+{
+    NPC_GRYSHKA         = 31433,
+    NPC_OLVIA           = 31425,
+    NPC_SANA            = 31429,
+    NPC_FELIKA          = 31427,
+    NPC_THUNG           = 31430,
+    NPC_KAJA            = 31423,
```
Comment (Goatform) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Remove comma.

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp (hunk 2 above)
Comment (tkrokli) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Sure, either like that or by just 2 slashes ( // ) in front of the text (more commonly used for commenting in 1 line).
/* and */ is more often used for multiple lines or a paragraph, like the Copyright text at the start of the files.

Comment (JamminWidYou) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: I meant whole script :P i do know how to comment stuff out

Comment (tkrokli) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Oh, right. If you want to comment out part of a script or a whole script to keep it from being part of the Travis compile, then it is correct if you add /* before the script starts and */ only at the end.

**Comment by JamminWidYou** [comment]
Done, let me know what needs to be fixed or rewritten

**Comment by tkrokli** [comment]
Suggestion for the ToDo list : use `- [ ] ` in front of the 2 entries where you have used StrikeThrough.
Wiki page for how to make them: https://github.com/blog/1825-task-lists-in-all-markdown-documents
`- [ ] ` for not yet done tasks, `- [X] ` for ticked boxes.

**Comment by msoky** [comment]
The Battle for the Undercity http://mmoarch.wikia.com/wiki/The_Battle_for_the_Undercity

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 7]
```diff
@@ -263,8 +264,563 @@ class npc_thrall_warchief : public CreatureScript
...
+    SAY_SYLVANAS_1              = 1, // As the combined Horde and Alliance forces began their assault upon the Wrath Gate, an uprising broke out in the Undercity. Varimathras and hordes of his demonic brethren attacked. Hundreds of my people were slain in the coup. I barely managed to escape with my life.
+    
+    SAY_JAINA_0                 = 0, // Thrall, what has happened? The King is preparing for war...
+    SAY_JAINA_1                 = 1, // I will deliver this information to King Wrynn, Thrall, but...
+    SAY_JAINA_2                 = 2, // Bolvar was like a brother to him. In the King's absence, Bolvar kept the Alliance united. He found strength for our people in our darkest hours. He watched over Anduin, raising him as his own.
+    SAY_JAINA_3                 = 3, // I fear that the rage will consume him, Thrall. I remain hopeful that reason will prevail, but we must prepare for the worst... for war.
+    SAY_JAINA_4                 = 4, // Farewell, Warchief. I pray that the next time we meet it will be as allies.
```
Comment (Goatform) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Remove comma.

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 8]
```diff
@@ -263,8 +264,563 @@ class npc_thrall_warchief : public CreatureScript
...
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_overlord_runthak_orgrimmarAI(creature);
+    }
+};
+
+/* -----------------------------------------------
+Phased out thrall during herald of war chain quest
```
Comment (Goatform) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Clean up this:
```
/* --------------------------------------------------------------
 Phased out thrall during herald of war chain quest
----------------------------------------------------------------*/
```
Into:

```
// Phased out thrall during herald of war chain quest

**Comment by JamminWidYou** [comment]
I will, need to fix few stuff first but a lil busy currently

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 9]
```diff
@@ -263,8 +264,563 @@ class npc_thrall_warchief : public CreatureScript
...
+{
+public:
+    npc_overlord_runthak_orgrimmar() : CreatureScript("npc_overlord_runthak_orgrimmar") { }
+
+    struct npc_overlord_runthak_orgrimmarAI : public ScriptedAI
+    {
+        npc_overlord_runthak_orgrimmarAI(Creature* creature) : ScriptedAI(creature) 
+        {
+            Initialize();
+            citizenList.clear();
```
Comment (jackpoz) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: this line is not needed because a class member std::list<T> is initialized empty by default

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 10]
```diff
@@ -263,8 +264,563 @@ class npc_thrall_warchief : public CreatureScript
...
+                            olvia->AI()->Talk(SAY_OLVIA_1);
+                        events.ScheduleEvent(EVENT_SCENE_3, 4500);
+                        break;
+                    case EVENT_SCENE_3:
+                        if (Creature* felika = ObjectAccessor::GetCreature(*me, felikaGUID))
+                            felika->AI()->Talk(SAY_FELIKA_1);
+                        events.ScheduleEvent(EVENT_SCENE_4, 4500);
+                        break;
+                    case EVENT_SCENE_4:
+                        if (Creature* thatung = ObjectAccessor::GetCreature(*me, thungGUID))
```
Comment (jackpoz) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: is it called thung or thatung ?

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 11]
```diff
@@ -263,8 +264,563 @@ class npc_thrall_warchief : public CreatureScript
...
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+    private:
+        EventMap events;
+        ObjectGuid guardsGUIDs[4];
```
Comment (jackpoz) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: please use a define or a const for the size of an array if you are going to use the size few times to ensure the correct size is always used

File: src/server/scripts/Spells/spell_quest.cpp [hunk 12]
```diff
@@ -2482,6 +2482,62 @@ class spell_q13665_q13790_bested_trigger : public SpellScriptLoader
         }
 };
 
+// herald of war
+class spell_59064_portal_orgrimmar : public SpellScriptLoader
```
Comment (Vincent-Michael) [inline src/server/scripts/Spells/spell_quest.cpp]: can spell_59064_portal_orgrimmar  / spell_59439_portal_undercity  in a gen script for both move?

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 13]
```diff
@@ -263,8 +264,563 @@ class npc_thrall_warchief : public CreatureScript
...
+};
+
+class areatrigger_orgrimmar_herald_of_war : public AreaTriggerScript
+{
+public:
+    areatrigger_orgrimmar_herald_of_war() : AreaTriggerScript("areatrigger_orgrimmar_herald_of_war") { }
+
+    bool OnTrigger(Player* player, AreaTriggerEntry const* /*trigger*/) override
+    {
+        if (player->IsAlive() && player->GetQuestStatus(QUEST_HERALD_OF_WAR) == QUEST_STATUS_COMPLETE && player->GetQuestStatus(QUEST_HERALD_OF_WAR) != QUEST_STATUS_REWARDED)
```
Comment (jackpoz) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: I see player->GetQuestStatus(QUEST_HERALD_OF_WAR) is checked against == QUEST_STATUS_COMPLETE and != QUEST_STATUS_REWARDED , is there a typo or is one of the checks not needed ?

File: src/server/scripts/Spells/spell_quest.cpp [hunk 14]
```diff
@@ -2482,6 +2482,62 @@ class spell_q13665_q13790_bested_trigger : public SpellScriptLoader
...
+public:
+    spell_59064_portal_orgrimmar() : SpellScriptLoader("spell_59064_portal_orgrimmar") { }
+
+    class spell_59064_portal_orgrimmar_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_59064_portal_orgrimmar_SpellScript);
+
+        void HandleScript(SpellEffIndex /*effIndex*/)
+        {
+            if (Unit* target = GetHitUnit())
```
Comment (Vincent-Michael) [inline src/server/scripts/Spells/spell_quest.cpp]: GetHitUnit cannot null

**Comment by jackpoz** [comment]
This PR has a few codestyle changes needed plus a couple of things to be modified, other than that it looks ready to be tested.
For the interested users: please test again the PR after it will be modified and provide a clear comment like "tested the whole event, works flawlessy"

File: src/server/scripts/Spells/spell_quest.cpp (hunk 12 above)
Comment (Goatform) [inline src/server/scripts/Spells/spell_quest.cpp]: Yeah, it should be converted into one script.

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp (hunk 10 above)
Comment (tkrokli) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Is this possibly related to Auctioneer Thathung (NPC ID 8673)?

File: src/server/scripts/Spells/spell_quest.cpp (hunk 14 above)
Comment (JamminWidYou) [inline src/server/scripts/Spells/spell_quest.cpp]: So i can do this then?
[code]
GetHitUnit()->CastSpell(GetHitUnit(), uint32(GetEffectValue()));
[/code]

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 15]
```diff
@@ -263,8 +264,597 @@ class npc_thrall_warchief : public CreatureScript
...
+        {
+            spawnedGuards = false;
+            Initialize();
+        }
+
+        void Initialize()
+        {
+            if (!spawnedGuards)
+            {
+                // clear guids only at first initialization before spawn. is it required to do so?
```
Comment (JamminWidYou) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: meant, 'before spawning guards' ;p

**Comment by JamminWidYou** [comment]
@jackpoz thanks for the feedback, i hope everything is alright now

**Comment by Killyana** [comment]
I scripted this 2 quests with Sai, I found that each quest use diff rent npcs and area trigger:
````
Alliance
SET @POSTWG := 58932;
SET @WGA_detectInvis := 60877;
SET @WGA_phase := 60815;
SET @WGH_phase := 59062;
SET @JAINA_SW := 32346;
SET @JAINA := 32364;
SET @THRALL := 32363;
SET @SYLV := 32365;
SET @GUARD := 32367;
SET @Broll := 32376;
SET @DUMMY := 32366; 
SET @PHASE := 128;
SET @PORTAL := 193948;        
SET @PORTAL_SW := 193207; 
SET @PORTAL_UC_A := 193955;
SET @TRIGGER := 5311;

Horde version:
SET @THRALL  := 31412;
SET @JAINA   := 31418;
SET @SYLV    := 31419;
SET @GRUNT   := 31416;
SET @KORKRON := 31417;
SET @Volgin  := 31649;
SET @TRIGGER := 5279;

**Comment by JamminWidYou** [comment]
@Killyana 
Thanks for the offer but no, less grid search this way and instead of having multiple scripts we have one, i believe its better than sai in this case

**Comment by Killyana** [comment]
If there's 2 events horde+alliance at the same time what happen?
Also I cannot find the script for @JAINA_SW := 32346;
``` sql
DELETE FROM `creature_text` WHERE `entry` IN (@JAINA_SW, @THRALL, @SYLV, @Jaina);
INSERT INTO `creature_text` (`entry`, `groupid`, `id`, `text`, `type`, `language`, `probability`, `emote`, `duration`, `sound`, `comment`, `BroadcastTextId`) VALUES
(@JAINA_SW, 0, 0, 'Do not do anything that would incite the Horde, $n. The Warchief has agreed to see us on good faith.', 12, 0, 100, 1, 0, 16124, 'Jaina Proudmoore',32799),
(@JAINA_SW, 1, 0, 'Let''s go.', 12, 0, 100, 1, 0, 16125, 'Jaina Proudmoore',32800),

**Comment by JamminWidYou** [comment]
As you said yourself, there is different entries for each event and alliance have their own phase so its different event alltogether and shold not collide if you script it, this PR is about a life without regret only ;p

**Comment by Killyana** [comment]
This is script is for the horde quest only?

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 16]
```diff
@@ -263,8 +264,579 @@ class npc_thrall_warchief : public CreatureScript
 
 };
 
+/* --------- Herald of War ------------- */
+enum citizenEntries
```
Comment (MitchesD) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: please name all enums Upper Camel Case

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 17]
```diff
@@ -263,8 +264,579 @@ class npc_thrall_warchief : public CreatureScript
...
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+    private:
+        EventMap events;
+        std::list<Unit*> citizenList;
```
Comment (MitchesD) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: oh come on, do not store pointers, use GUIDs or in this case it looks like local variable should be enough

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 18]
```diff
@@ -263,8 +264,579 @@ class npc_thrall_warchief : public CreatureScript
...
+        npc_overlord_runthak_orgrimmarAI(Creature* creature) : ScriptedAI(creature) 
+        {
+            Initialize();
+        }
+
+        void Initialize() 
+        {
+            inProgress = false;
+            me->GetMotionMaster()->MovePath(me->GetSpawnId() * 10, true);
+            if (citizenList.empty())
```
Comment (MitchesD) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: why do you need this condition? you are calling Initialize() when event is done, but in this time that list isn't empty so I think this scope will never be used, and on initialization all the GUIDs are empty, so what's the purpose?

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 19]
```diff
@@ -263,8 +264,579 @@ class npc_thrall_warchief : public CreatureScript
...
+            {
+                gryshkaGUID.Clear();
```
Comment (MitchesD) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: also I wonder why you need this, these NPCs look like static, they can't be dead or something else, so I think you don't need it

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 20]
```diff
@@ -263,8 +264,579 @@ class npc_thrall_warchief : public CreatureScript
...
+        {
+            if (who->GetTypeId() == TYPEID_PLAYER && who->IsWithinDist(me, 20.0f) && !inProgress)
+            {
+                inProgress = true;
+                if (citizenList.empty())
+                {
+                    Trinity::AnyFriendlyUnitInObjectRangeCheck checker(me, me, 25.0f);
+                    Trinity::UnitListSearcher<Trinity::AnyFriendlyUnitInObjectRangeCheck> searcher(me, citizenList, checker);
+                    me->VisitNearbyObject(20.0f, searcher);
+                    for (std::list<Unit*>::iterator itr = citizenList.begin(); itr != citizenList.end(); ++itr)
```
Comment (MitchesD) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: use range-based loop and also try to make it a little bit easier

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 21]
```diff
@@ -263,8 +264,579 @@ class npc_thrall_warchief : public CreatureScript
...
+                spawnedGuards = true;
+                for (uint8 i = 0; i < GUARD_GUIDS_SIZE; i++)
+                {
+                    if (Creature* korkronGuard = me->SummonCreature(NPC_KORKRON_GUARD, guardsSpawnPosition[i]))
+                        guardsGUIDs[i] = korkronGuard->GetGUID();
+                }
+            }
+        }
+
+        void Reset() override
```
Comment (MitchesD) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: you don't need to reset EventMap as well?

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp (hunk 18 above)
Comment (JamminWidYou) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Thought we need to clear em first like we used to with uint64, never used them before. I'll fix it later, thanks for the feedback

**Comment by OScriptTC** [comment]
What is needed for this PR? :-)

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 22]
```diff
@@ -263,8 +264,557 @@ class npc_thrall_warchief : public CreatureScript
...
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+
+    private:
+        EventMap events;
+        bool inProgress;
```
Comment (jackpoz) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: this field is not currently initialized in the constructor, please initialized it

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 23]
```diff
@@ -263,8 +264,557 @@ class npc_thrall_warchief : public CreatureScript
...
+};
+
+Position const MiscMovePositions[3] =
+{
+    { 1921.719604f, -4143.051270f, 40.623356f, 1.657789f }, // jaina move
+    { 1921.151855f, -4139.343750f, 40.583084f, 4.732627f }, // thrall move
+    { 1918.732422f, -4139.619629f, 40.607685f, 4.803311f } // sylvanas move
+};
+
+const uint32 GUARD_GUIDS_SIZE       = 4;
```
Comment (jackpoz) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: this should be used as size for every array related to guards, like GuardsSpawnPosition and GuardsMovePosition

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp [hunk 24]
```diff
@@ -263,8 +264,557 @@ class npc_thrall_warchief : public CreatureScript
...
+        }
+
+    private:
+        EventMap events;
+        ObjectGuid guardsGUIDs[GUARD_GUIDS_SIZE];
+        ObjectGuid jainaGUID;
+        ObjectGuid sylvanasGUID;
+        ObjectGuid stormwindPortalGUID;
+        bool spawnedGuards;
+        bool sceneInProgress;
```
Comment (jackpoz) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: this field should be initialized in the constructor

Comment (JamminWidYou) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Thanks, i'll fix it this weekend. Out of town currently ;p

**Comment by JamminWidYou** [comment]
holy crap that squashing was stressful, i hope i did everything alright

**Comment by tkrokli** [comment]
As far as squashing the commits into 1, it worked just OK.
I did not notice if the compile error was there before, though.

**Comment by JamminWidYou** [comment]
yeah yeah im stupid ;p, i'll fix that when im at PC in few hours

**Comment by JamminWidYou** [comment]
Btw this is 3.3.5a only - they removed that chain in cataclysm

**Comment by tkrokli** [comment]
Yes, that is properly tagged with the green label [Branch-3.3.5a] and the title prefix [3.3.5]

**Comment by JamminWidYou** [comment]
I dont want to be a dick with bumping PRs, but is something wrong with this PR that i need to fix or just have to wait for someone to review it again and merge if he thinks its okay?

**Comment by DDuarte** [comment]
@JamminWidYou: No problem. Currently most PRs are on hold, until the cherry-picks/merge between branches is finished and the next TDB is released. Once that's done, PRs will be *unlocked*.

(We probably should have a roadmap documented somewhere)

**Comment by Treeston** [comment]
Two things I noticed skimming your SQL:
- Please use `@CGUID` for the first free creature guid you need and `@OGUID` for the first free object guid you need.
- Also add a comment indicating how many GUIDs will be used to the `SET` line (see [this](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/sql/updates/world/2016_03_23_00_world.sql#L1-L14) for reference).

Both of those make finding a free GUID block for the PR far easier for the person performing the merge. Thanks!

File: src/server/scripts/Kalimdor/zone_orgrimmar.cpp (hunk 24 above)
Comment (Kittnz) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: Nothing changed? @JamminWidYou

Comment (jackpoz) [inline src/server/scripts/Kalimdor/zone_orgrimmar.cpp]: the field looks initialized now

**Comment by JamminWidYou** [comment]
@Treeston 
aight will do

**Comment by Aokromes** [comment]
Any reason to don't merge this?

**Comment by JamminWidYou** [comment]
Done on my end, since you can squash via github now i won't bother to squash :P

**Comment by tkrokli** [comment]
Please use the table column names in your `INSERT INTO` queries (TC codestyle).

- **https://trinitycore.atlassian.net/wiki/display/tc/SQL+Guidelines**

(Section: **Delete before insert into and always put fields in the query.**)

**Comment by JamminWidYou** [comment]
Now you're telling me about this, after 6 months? :|

**Comment by tkrokli** [comment]
Sorry about that, it did not occur to me before I had to comment on the same in a recent SQL issue where DDuarte/Nayd commented on this in #trinity regarding that new issue.

**Comment by mik1893** [comment]
is this all good? can we merge?

**Comment by JamminWidYou** [comment]
Unless someone has any objections then yeah, merge ready

File: sql/updates/world/life_without_regrets.sql [hunk 25]
```diff
@@ -0,0 +1,385 @@
+SET @CGUID:= 213733; -- 69 GUIDs required
```
Comment (Aokromes) [inline sql/updates/world/life_without_regrets.sql]: before merge remember to search for holes on guids

**Comment by mik1893** [comment]
@Aokromes search for a hole in guids plz :dancer:

File: sql/updates/world/life_without_regrets.sql (hunk 25 above)
Comment (tkrokli) [inline sql/updates/world/life_without_regrets.sql]: Based on current content in the `creatures` table in TDB 335.61 + updates up to and including `2016_06_14_02_world.sql` , the first 69 consecutive free GUIDs are from `145427` to `145495` (including both). 

Search tool: Unused Guid Searcher

**Comment by tkrokli** [comment]
In the `creature` table in TDB 335.61 + updates up to and including `2016_06_14_02_world.sql` , the first 69 consecutive free GUIDs are from 145427 to 145495 (including both).

**Comment by Treeston** [comment]
`creature (69 GUIDs):` 145427-145495
`gameobject (13 GUIDs)`: 73587-73599

**Comment by mik1893** [comment]
ok ill merge and edit the sql then

**Comment by mik1893** [comment]
nice work @JamminWidYou , keep it up - promise next PR won't take the same amount of time!!!

**Comment by Aokromes** [comment]
Now you can reclaim the bounty :)

**Comment by zaphod77** [comment]
so is this just horde side, or are both done now?

the two are quite different.

And how much is finished?  

mad props if this is the whole quest.

I did make something before starting with battle at lights hope chapel as the base code, and it was a reasonable approximaionexcept i had pathing issues i coulldn't solve at the end and boss A.I. wasn't 100%.  

pretty impressed if this really done.

**Comment by tkrokli** [comment]
@zaphod77 : https://github.com/TrinityCore/TrinityCore/pull/16067#issuecomment-170373915 :: [...] this PR is about A Life Without Regret only [...]
- http://wow.gamepedia.com/Quest:A_Life_Without_Regret
- http://www.wowhead.com/quest=13266/a-life-without-regret

**Comment by dr-j** [comment]
PR also fixed http://www.wowhead.com/quest=13257/herald-of-war, chain still broke at http://www.wowhead.com/quest=13369/fate-up-against-your-will for alliance players though.
